google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java

```diff
@@ -573,6 +573,9 @@ private void requestCallback(AppendRowsResponse response) {
         conectionRetryCountWithoutCallback = 0;
       }
       requestWrapper = pollInflightRequestQueue();
+      if (requestWrapper == null) {
```

Comment:
If we haven't cleaned up and this happens, we should throw an exception

Maybe instead of using null, check if we've cleaned up, and don't call the method (since this is under lock anyways)?

Reply:
If we have cleaned up, then all inflight requests are all set with an exception already. 
I don't think we should throw on requestCallback?

Reply:
I meant the opposite: if we haven't cleaned up.  

If during normal operations we get a callback but there's nothing in the queue, that should never happen, so we should throw (or somehow error out)

Reply:
Updated the code, hope it is more straightforward.

Reply:
Yeah it is, but I was thinking we'd also have

onDone {
  this.done = true;
}

and in your change:
if (!this.done && this.inflightRequestQueue.isEmpty()) {
  // Handle this error case!
}

But since it's in the callback, that's probably more complicated than I originally thought... Should we handle that in a followup?

Reply:
PTAL

Reply:
There is no super good place for done, since the lock is done inside of the cleanupInflightQueue.

Reply:
Added an error message, without a response, user will probably not be able to get the actual exception.